website/database.py

```diff
@@ -109,7 +109,8 @@ def __init__(self):
                                   }),
                                   indexes=[
                                       dynamo.Index('email'),
-                                      dynamo.Index('epoch', sort_key='created')
+                                      dynamo.Index('epoch', sort_key='created'),
+                                      dynamo.Index('username', sort_key='epoch', keys_only=True)
```

Comment:
No please.

```suggestion
                                      dynamo.Index('epoch', sort_key='username', keys_only=True)
```

If the DynamoDB layer allowed you to formulate the query below, something is wrong ☹️

Reply:
Oh I see, the abstraction layer complains if you use the same partition key multiple times. That's not strictly necessary, but for convenience. Let me see what I can do about that.

Reply:
Still, when I try to set up a table with these indexes and run that `get_many` query, I get an error:

```
Non-Equals conditions only allowed on sort key sort, got: ['username']
```

Reply:
Here is a PR you will need to merge before you can set up the indexes the way you need them: https://github.com/hedyorg/hedy/pull/6034. Unfortunately, this will also require you to annotate all existing epoch queries to indicate what index to use. I know at least one of them in the admin interface, and there aren't any tests on those, so do have a good look around.

Reply:
> Oh I see, the abstraction layer complains if you use the same partition key multiple times so you were forced to reverse the keys to get it to do anything at all.
> 
> That restriction is not strictly necessary. It's not enforced by Dynamo, but by our abstraction layer for the convenience of identifying which index to query. Let me see what I can do about that.

Yes, so I thought I was making a mistake. Thanks for pointing it out and fixing it.

> (BTW I created the necessary (epoch, username) indexes in the actual DDB tables already, so you don't need to do anything there anymore)

Neat!